src/libutil/serialise.hh

```diff
@@ -368,6 +370,10 @@ Sink & operator << (Sink & sink, std::string_view s);
 Sink & operator << (Sink & sink, const Strings & s);
 Sink & operator << (Sink & sink, const StringSet & s);
 
+inline Sink & operator << (Sink & sink, const std::optional<std::chrono::microseconds> & duration) {
+    return sink << (duration.has_value() ? duration.value().count() : -1l);
+}
+
```

Comment:
Please don't add more functionality here, this should go in `worker-proto.cc` for just that one, or `common-proto.cc` to use in both (e.g. in a follow-up PR for that one if we decide to do it you can move it there)

While `-1` is an unlikely duration to occur in practice, I would use an optional tag instead see 57d960dcd174f0910c7fca02ee3afe80cb2b7844 for example.

Reply:
But that commit removes the tags in favour of using an empty string for the null option, right?

You meant that you'd prefer that we use an empty string as well, or that we re-introduce tags to distinguish between the two variants?

I used strings initially, but then went for integers to have a more efficient binary representation.

Reply:
Oh, I saw how optional is dealt with in other serialisers in worker-protocol.cc, so I'll do it similarly. Clearly I'm still getting to know the codebase. Thanks for the pointers!